modules/redis/scanner.go

```diff
@@ -228,44 +362,97 @@ func (scanner *Scanner) Scan(target zgrab2.ScanTarget) (zgrab2.ScanStatus, inter
 	}
 	defer scan.Close()
 	result := scan.result
-	pingResponse, err := scan.SendCommand("PING")
+	pingResponse, err := scan.SendCommand(scanner.commandMappings["PING"])
 	if err != nil {
-		// if the first command fails (as opposed to succeeding but returning an
+		// If the first command fails (as opposed to succeeding but returning an
 		// ErrorMessage response), then flag the probe as having failed.
 		return zgrab2.TryGetScanStatus(err), nil, err
 	}
 	// From this point forward, we always return a non-nil result, implying that
 	// we have positively identified that a redis service is present.
 	result.PingResponse = forceToString(pingResponse)
 	if scanner.config.Password != "" {
-		authResponse, err := scan.SendCommand(scanner.config.AuthCommand, scanner.config.Password)
+		authResponse, err := scan.SendCommand(scanner.commandMappings["AUTH"], scanner.config.Password)
 		if err != nil {
 			return zgrab2.TryGetScanStatus(err), result, err
 		}
 		result.AuthResponse = forceToString(authResponse)
 	}
-	infoResponse, err := scan.SendCommand("INFO")
+	infoResponse, err := scan.SendCommand(scanner.commandMappings["INFO"])
 	if err != nil {
 		return zgrab2.TryGetScanStatus(err), result, err
 	}
 	result.InfoResponse = forceToString(infoResponse)
-	infoResponseBulk, ok := infoResponse.(BulkString)
-	if ok {
+	if infoResponseBulk, ok := infoResponse.(BulkString); ok {
+		fieldsWeAreLookingFor := 12
+		fieldsFound := 0
 		for _, line := range strings.Split(string(infoResponseBulk), "\r\n") {
-			if strings.HasPrefix(line, "redis_version:") {
-				result.Version = strings.SplitN(line, ":", 2)[1]
+			switch line_prefix_suffix := strings.SplitN(line, ":", 2); line_prefix_suffix[0] {
+			case "redis_version":
+				fieldsFound += 1
+				result.Version = line_prefix_suffix[1]
```

Comment:
the second argument to `SplitN` sets a maximum number of array entries, but not the minimum -- so something like
```
arch_bits: 64
os
redis_build_id: xyz abc
```

would lead to an array index out of bounds panic, since `len(line_prefix_suffix)` on the line without a colon would be 1, not 2. This worked before because it had already ensured that there was at least one colon in the string before calling `SplitN`.

Reply:
Great catch, thank you! I added a check to see if `len(line_prefix_suffix) == 1` then I append an empty string